### ISSUE TITLE
Fix unbound memory allocation in load

### ISSUE THREAD

**Issue by eberleim** [gridlab-d/gridlab-d#1228]
Initial Checklist prior to submitting the issue. Check the bellow boxes to the best of your knowledge:

- [X] I have not found an open issue describing the same problem.
- [X] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [X] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
The `load_set_index` function in gldcore/load.c currently dynamically allocates two large arrays of type `OBJECT *` (8-bytes) and `unsigned char` (1-byte). By default, each array is 65k elements, however user input for object-id is used to determine the upper bound (as noted in #1225), allowing users to instantiate arbitrarily large object IDs `object meter:2010000000 {...}` which force GridLAB to produce equally large, mostly empty heap-allocated memory blocks. 
For sufficiently large IDs, this causes an out-of-memory exception. 
This behavior is inefficient and undesirable, however the current C-implementation would be extremely hard to retrofit with more correct memory behavior. 

*Note: This issue is created as a stub until the C++ refactor is merged into the develop branch, as the C++ implementation will make this problem significantly easier to solve.*
Depends on #1091

-------------------------------------------------------------------------------------------------------------

## Possible C++ solution: 

C++ object wrapper containing required object_linked logic and `std::unordered_map<size_t, OBJECT*>` with operator[] overload for `size_t` input, returning `OBJECT*` from map. `std::unordered_map` will be size-constrained to at most 2*elements, and will automatically resize as appropriate, so arbitrarily large IDs can be used without causing memory allocations.
